### PR TITLE
Adds Inbelly Spawning ability

### PR DESCRIPTION
**Pull request by Heroman3003**
Allows prey to join the round directly into predator's insides! A tool that properly enables longterm and/or multiround scenes of various kinds. Available in Ghost tab to anyone, requires pred to have a new vore panel preference enabled (disabled by default).

More details:

Adds new vore panel preference - Inbelly Spawning. When enabled allows you to recieve requests to be spawned inbelly. Disabled by default.

Prey can use Ghost tab verb Spawn In Belly to get a list of eligible preds.
Preds are eligible if they are a player-controlled vore-capable humanmob, simplemob or siliconmob (except AI) that is currently located on one of the station's (or ship's I guess) main levels.

If prey selects a pred, pred gets a series of prompts.
First prompt is notifying them of prey's ckey and selected character slot's name, and asking if they accept request or not.
If they do accept it, they are prompted to select a belly. If a belly is currently on Digest mode, additional confirmation is requested.
After selecting a belly pred can optionally choose to have prey start pre-absorbed.
After all that, pred is given a short brief of the result as final confirmation.

If they confirm, prey is given same brief and opportunity to confirm.
Should both pred and prey confirm positively, spawning happens.

Prey is spawned naked, without any equipment. They recieve no records or manifest entry, and their arrival is not announced over comms. They are not backup-implanted or anything like that either.

All Inbelly Spawns are admin logged and notified, and using the verb brings up a warning, similarly to translocator, notifying of what this verb is and what its intended purpose is, as well as giving an anti-abuse warning.

That should about sum it up.

### CODE REVIEW

**Comment by Screemonster** [comment]
yeshahaha

**Comment by Haery70** [comment]
Have been wanting this idea for years, though something that I wanted to bring to attention regarding this PR: How does it handle players that have already joined the round and died? Do they have the option to respawn inside a pred or are they locked out of this feature? I can already see potential abuse value for away teams not just from exploration as it could bypass resleeving entirely to get an easy fixup despite the fact that it's going to be adminlogged regardless. I feel like that this shouldn't apply to those that have already perished in non-vore-related means and I'd hate to see such a cool feature get struck down for potential metagaming reasons.

**Comment by Heroman3003** [comment]
I'll leave whether any additional restrictions are needed up to staff choice, but I will say that leaving possibility of spawning after dying in round wasn't really unintentional. The verb has proper warning and if someone straight up ignores it to powergame... Well, combination of it only being doable when on station_levels and the fact that admins are actively notified when someone does that, I do think it shouldn't be feasible.

File: code/modules/vore/eating/inbelly_spawn.dm [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+				continue
+			if(!SM.allow_inbelly_spawning)
+				continue
+			eligible_targets += SM
+			continue
+
+		// Only humans, simple_mobs and non-AI silicons are included. Obscure stuff like bots is skipped.
+
+	if(!eligible_targets.len)
+		to_chat(src, "<span class=notice>No legible preds were found.</span>")				// :(
```
Comment (Very-Soft) [inline code/modules/vore/eating/inbelly_spawn.dm:54]: legible should probably be eligible

Comment (Heroman3003) [reply code/modules/vore/eating/inbelly_spawn.dm]: Fixed.

File: code/modules/vore/eating/inbelly_spawn.dm [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+	if(!target || !client)			// Did out target cease to exist? Or did we?
+		return
+
+	// Notify them that its now pred's turn
+	to_chat(src, "<span class=notice>Inbelly spawn request sent to predator.</span>")
+	target.inbelly_spawn_prompt(client)			// Hand reins over to them
+
+/mob/living/proc/inbelly_spawn_prompt(client/potential_prey)
+	if(!potential_prey || !istype(potential_prey))		// Did our prey cease to exist?
```
Comment (BlackMajor) [inline code/modules/vore/eating/inbelly_spawn.dm:67]: `[potential_prey.ckey]`
Probably shouldn't dox ckeys of players to eachother, some players would rather hide that.

Comment (Heroman3003) [reply code/modules/vore/eating/inbelly_spawn.dm:67]: Considering this is a kind of feature that requires at least a certain level of OOC coordination to make work in the first place, this isn't that problematic. This isn't meant ot be a valid anonimous latejoin method.

**Pull request merged**